### DISCUSSION TITLE
CI/CD with custom Base Application

### DISCUSSION THREAD

**Discussion by ShinaziFarukiMSD** [microsoft/AL-Go#1576] [Q&A]
Hello,

i have a custom base application and need to use it for CI/CD process. Which is the better way to do this?

**Comment by freddydk** [comment]
AL-Go for GitHub doesn't support modified base applications out of the box, it requires some configuration.
https://github.com/microsoft/BCApps is the system app, in which we override the NewBcContainer function to remove all apps before deploying - you would need to do something similar if you want to compile the base app.
We also have the private preview here: https://github.com/microsoft/BusinessCentralApps in which we build the base application in AL-Go for GitHub, but without running any tests.
Alternatively you could look at ALOps (which is for Azure DevOps), Waldo might have templates for building modified base applications.

**Comment by LukasSchauffele** [comment]
Hi @freddydk - we are also interested in compiling a custom base app as part of our build pipelines. The link to the private preview is giving me a 404 - not found. Is there another form I can fill out to join the private preview?

**Comment by freddydk** [comment]
@JesperSchulz should be able to help you there I think

**Comment by JesperSchulz** [comment]
That I can!
For access to the pilot, please fill out the following form: https://forms.office.com/r/E9wyE5a58f.
